[Chuang Zhe Quan] iP

ChuangZheQuan opened the pull request:
# DukePro
DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is, 
1. download it from [here](https://www.youtube.com/watch?v=Zd8vzIRQLLM&ab_channel=springgrapevine).
2. double-click it.
3. add your tasks
4. let it manage your tasks for you 😉

And it is **FREE**!

Features: 
- [ ] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method: 
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ChuangZheQuan committed:
Level 1 commit

ChuangZheQuan committed:
Level 2 commit

ChuangZheQuan committed:
Level 2 commit

ChuangZheQuan committed:
Level 3 commit

ChuangZheQuan committed:
Level 4 commit

ChuangZheQuan committed:
A-TextUiTesting Commit

ChuangZheQuan committed:
Level 5 Commit

ChuangZheQuan committed:
Level-6 commit

ChuangZheQuan committed:
Level 3 Commit

ChuangZheQuan committed:
Merge commit 'f279e40bba87c0605a7f23c42bc009053004d494'

* commit 'f279e40bba87c0605a7f23c42bc009053004d494':
  Level-6 commit
  Level 5 Commit
  A-TextUiTesting Commit
  Level 4 commit
  Level 3 commit
  Level 2 commit

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Task.java

ChuangZheQuan committed:
Add DukeHandler to refactor Duke

ChuangZheQuan committed:
Refactor other methods to DukeHandler

ChuangZheQuan committed:
Update FileHandler and add load, write features

ChuangZheQuan committed:
Add done delete functionality to FileHandler

ChuangZheQuan committed:
Add Date Time for Level-8

ChuangZheQuan committed:
Merge branch 'branch-level-8'

* branch-level-8:
  Add Date Time for Level-8

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

ChuangZheQuan committed:
Implement More OOP

ChuangZheQuan committed:
Implement A-Packages

ChuangZheQuan committed:
Add JUnit tests

ChuangZheQuan committed:
Add JUnit Tests

ChuangZheQuan committed:
Create a JAR File

ChuangZheQuan committed:
Add JavaDoc to public methods

ChuangZheQuan committed:
Abide by coding standard

ChuangZheQuan committed:
Add find to parser

ChuangZheQuan committed:
Add find test

ChuangZheQuan committed:
Change constants to uppercase

ChuangZheQuan committed:
Follow coding conventions

ChuangZheQuan committed:
Fix coding standard

ChuangZheQuan committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Add Gradle support

ChuangZheQuan committed:
Add Checkstyle

ChuangZheQuan committed:
Implement Level-10

ChuangZheQuan committed:
Clear user input

XXJJXJ commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,92 @@
+
+
+public class Storage {
+    private final String FILENAME;
+    private final TaskList TASKS = new TaskList(new ArrayList<>());
+
+    Storage(String FILENAME) {
+        this.FILENAME = FILENAME;
```
Are FILENAME and TASKS really considered constants?

XXJJXJ commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,24 @@
+package duke;
+
+public class Ui {
+//    public static void print(String input) {
+//        System.out.println("\t" + input);
+//    }
```
I guess the identation of your comment here does not fit the coding standards

XXJJXJ reviewed (commented):
Generally good coding standards! Just missing some javadocs in a few files and a few minor problems.

ChuangZheQuan committed:
Fix styles

ramaven commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,51 @@
+
+public class DukeTest {
+    @Test
+    public void addTask_event_success(){
```
perhaps you could name this method in camel casing as addTaskEventSuccess(), and same for removeTaskEventSuccess() and findTaskEventSuccess()

ramaven commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,16 @@
+package duke;
+
+public class Todo extends Task {
+    Todo(String desc) {
```
Perhaps you could name the argument as 'description'  instead of 'desc' so it is more intuitive, but 'desc' is alright too!

ramaven reviewed (commented):
Overall, your code is very well-organised and readable; Good Job! :D

ChuangZheQuan committed:
Commit A-Assertions

ChuangZheQuan committed:
Commit A-CodeQuality

Some methods exceeds 30 lines of code

This may affect user readability.

Let's,
* shorten the methods as much as possible
* extract methods out of long methods

ChuangZheQuan committed:
Merge pull request #1 from ChuangZheQuan/branch-A-Assertions

Commit A-Assertions

ChuangZheQuan committed:
Merge pull request #2 from ChuangZheQuan/branch-A-CodeQuality

branch-A-CodeQuality

ChuangZheQuan committed:
Add first snooze commit

ChuangZheQuan committed:
Snooze: implement Snooze feature

There is no Snooze feature.

Snooze feature can allow users to change their schedule

Let's,
* add a command to detect snoozes
* add a script to change the date and time of deadlines and events

ChuangZheQuan committed:
Merge branch 'branch-Snooze'

* branch-Snooze:
  Snooze: implement Snooze feature
  Add first snooze commit

ChuangZheQuan committed:
Snooze: implement Snooze feature

There is no Snooze feature.

Snooze feature can allow users to change their schedule

Let's,
* add a command to detect snoozes
* add a script to change the date and time of deadlines and events

ChuangZheQuan committed:
Improve Code Quality

ChuangZheQuan committed:
Make GUI better

ChuangZheQuan committed:
Improve Code Quality

ChuangZheQuan committed:
Fix parse bug and create Ui.png

ChuangZheQuan committed:
Set theme jekyll-theme-tactile

ChuangZheQuan committed:
Add bye command and write user guide

ChuangZheQuan committed:
Merge branch 'master' of https://github.com/ChuangZheQuan/ip

ChuangZheQuan committed:
Update user guide

ChuangZheQuan committed:
Update user guide

ChuangZheQuan committed:
Update user guide

ChuangZheQuan committed:
Fix user guide

ChuangZheQuan committed:
Fix UG

ChuangZheQuan committed:
Add more exception handling

ChuangZheQuan committed:
Update README to include link to jar file

ChuangZheQuan committed:
Handle more errors for deleting task

ChuangZheQuan committed:
Improve java docs and update readme

ChuangZheQuan committed:
Update readme
